source/AI.cpp

```diff
@@ -651,12 +651,33 @@ void AI::Step(const PlayerInfo &player, Command &activeCommands)
 			continue;
 		}
 
-		// Run away if your hostile target is not disabled and you are badly damaged.
+		// Check if a ship is armed with only ammo-using weapons,
+		// but no longer has the ammunition needed to use them.
+		bool neeedsAmmo = false;
+		for(const Hardpoint &hardpoint : it->Weapons())
+		{
+			const Weapon *weapon = hardpoint.GetOutfit();
+			if(weapon && !hardpoint.IsAntiMissile())
+			{
+				const Outfit *ammo = weapon->Ammo();
+				if(!ammo || it->OutfitCount(ammo))
+				{
+					// This ship has at least one usable weapon.
+					neeedsAmmo = false;
+					break;
+				}
+				else
+					neeedsAmmo = true;
+			}
+		}
+
+		// Run away if your hostile target is not disabled
+		// and you are either badly damaged or have run out of ammo.
 		// Player ships never stop targeting hostiles, while hostile mission NPCs will
 		// do so only if they are allowed to leave.
 		const bool shouldFlee = (personality.IsFleeing() ||
 			(!personality.IsDaring() && !personality.IsStaying()
```

Comment:
The daring personality check should only apply to the low health check. If a daring ship needs ammo it should IMO still flee.